hop/auth.py

```diff
@@ -5,26 +5,48 @@
 
 import argparse
 import errno
+import getpass
 import logging
 import os
 
+import toml
+
 from adc import auth
 
 DEFAULT_AUTH_CONFIG = """\
-security.protocol=SASL_SSL
-sasl.mechanism=SCRAM-SHA-512
-sasl.username=username
-sasl.password=password
+[auth]
+username = "{username}"
+password = "{password}"
 """
 
 logger = logging.getLogger("hop")
 
-# expose auth options from adc
-SASLAuth = auth.SASLAuth
 SASLMethod = auth.SASLMethod
 
-# map default auth option to Auth
-Auth = auth.SASLAuth
+
+class Auth(auth.SASLAuth):
```

Comment:
Not much, the only intended purpose is to set the default method to be `SCRAM-SHA-512` instead of the default in `adc-streaming`. This could be the default in `adc-streaming` or maybe there's an easier way to do this if we wanted to decouple them in any way.

Reply:
Ah, yeah, I see. What you have here makes some sense. You could do it as a function too (`def default_auth` or something), but I'm not sure that's much better.

Could you add a little comment explaining that this class exists to set our desired defaults? I have been thinking a lot lately about https://brooker.co.za/blog/2020/06/23/code.html and it's inspired me to push for comments which describe *design decisions*, rather than just comments that describe what code does, and this is a good example.

Reply:
Thanks @spenczar, that was a good read and a nice approach to take for documenting code. I added some reasoning above the class definition. Let me know if that sounds good or whether you'd prefer it somewhere else (docstring or otherwise).